### DISCUSSION TITLE
Best way to log every request

### DISCUSSION THREAD

**Discussion by valankar** [zauberzeug/nicegui#4073] [Q&A]
### Question

I have some pages setup with the ui.page decorator:

```python
@ui.page("/")
def foo():
  log_request()
  ...
```

I'd like to log something for all pages. I've been adding my log_request() call to every page method, but I'm wondering if there is a better way to do this without repeating it? I tried to create a decorator wrapping ui.page but could not get it to work.

**Comment by rodja** [comment]
We have a few techniques demonstrated in https://github.com/zauberzeug/nicegui/tree/main/examples/modularization/.
